src/compiler/checker.ts

```diff
@@ -8091,13 +8091,22 @@ namespace ts {
             return expression;
         }
 
+        function getControlFlowContainer(node: Node): Node {
+            while (true) {
+                node = node.parent;
+                if (isFunctionLike(node) || node.kind === SyntaxKind.ModuleBlock || node.kind === SyntaxKind.SourceFile || node.kind === SyntaxKind.PropertyDeclaration) {

```

Comment:
Does this cover computed properties?

```ts
const LANG = "German";
class WurstLanguage {
    [LANG]() {
    }
}
```

Maybe the class itself should be a control flow container.